chore(stock movements registry): filter movement by user

mbayopanda opened the pull request:
This PR add a filter for users, so that we can perform also search stock movements by users.

mbayopanda committed:
filter movement by user

jniles commented on client/src/modules/stock/movements/modals/search.modal.js:
```diff
@@ -8,46 +8,52 @@ SearchMovementsModalController.$inject = [
 
   // map key to last display value for lookup in loggedChange
-  var lastDisplayValues = lastViewFilters.reduce(function (object, filter) {
+  lastDisplayValues = lastViewFilters.reduce(handleLastViewFilter, {});
```
Most other registries put this on the service - see [the Patients search modal](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/invoices/registry/search.modal.js#L24) for example.  The method is `getDisplayValueMap()` on the FilterService.

jniles reviewed (commented):
Code looks fine to me.  Could you include an integration test to make sure the API always works?

jniles committed:
test(stock): add movements user filter test

This commit adds an integration test for the user_id filter.  It also
improves the search modal code slightly.
